components/config/src/config/markup.rs

```diff
@@ -136,6 +136,20 @@ impl Markdown {
             }
         }
 
+        // Validate that all exported highlight themes exist as well
+        for theme in self.highlight_themes_css.iter() {
+            let theme_name = &theme.theme;
+            if !(THEME_SET.themes.contains_key(theme_name)) {
+                // Check extra themes
+                if let Some(extra) = &*self.extra_theme_set {
```

Comment:
You can probably do something like:

```
if self.extra_theme_set.map(|e| extra.themes.contains_key(theme_name)).unwrap_or(false) {
   bail!(...)
}
```

Reply:
I'm a little confused on how this cleans things up -- imo its a little cleaner to unwrap the `self.extra_theme_set` and _then_ check if it contains our specified element; how does the `.map` work here?

Reply:
It will run what's in the `map` only if `self.extra_theme_set` is not `None`. Unless i missed something it should be equivalent to your code. It's up to personal preference after that :)